src/ModelContextProtocol.AspNetCore/SseHandler.cs

```diff
@@ -97,7 +96,7 @@ public async Task HandleMessageRequestAsync(HttpContext context)
             return;
         }
 
-        var message = (JsonRpcMessage?)await context.Request.ReadFromJsonAsync(McpJsonUtilities.DefaultOptions.GetTypeInfo(typeof(JsonRpcMessage)), context.RequestAborted);
+        var message = await StreamableHttpHandler.ReadJsonRpcMessageAsync(context);
```

Comment:
Might be cleaner to move this method to its own type that can be referenced by both.

Reply:
We already reference StreamableHttpHandler static methods a lot in SseHandler. AIFunctionMcpServerTool.DeriveName(...) is similarly shared between the other AIFunction derived types for prompts and resources. When it's only a few things, it's not worth creating a new static "utility" class imo.